### PR TITLE
Adds the bluespace crusher, a new arm implant

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: The bluespace crusher arm implant can now be built in robotics and R&D. It has the power to erase matter and space!
/:cl:

## What does it do
Before or after implanting, it must be loaded with bluespace crystals to get charges. Every crystal can store a charge and there can be up to 3 crystals.
Regenerates 1 charge every 60 seconds; any use of this item requires a charge.
### When attacking people in melee
Deals 30 completely armor-piercing damage.
### When attacking items in melee
Deletes the item.
### When attacking other objects (structures mostly)
Deals 150 damage, like hulks.
### When using it at range
Destroys space between you and the target, rapidly moving it and everything on the same turf towards you. Will stop moving if it hits an obstacle. Won't move anchored objects/mobs.
### Blocking a thrown item
75% chance of destroying the item mid-flight. Uses and requires a charge, so it can be abused to drain someone's implant.

## Other Changes
I fixed a bug where the throwing's check shield proc call did not pass the proper attack flag. This should probably help with other throw blocking items' bugs, but i didn't test on them.
Hit_reaction now also has the attacking atom as an argument. Check shields already had it, but never used it.

## Example Gif
https://i.gyazo.com/95c79bb10523d6f6f1c2062e2f44f13a.gif

## Icons (they're kinda shitty, could really use a real spriter's work)
Implant: crusher
Item: hando

## But why
It's a gimmick science item that is not simply an upgrade over other weapons.
It's powerful, bit it's also expensive, has a very long cooldown and requires surgical implantation to work, i doubt it'll be overpowered.

### CODE REVIEW

**Comment by InsaneHyena** [comment]
It's the stand even my big bro fears, ZA HANDO.

**Comment by Kromgar** [comment]
image



IS THIS A MOTHAFUCKIN JOJO REFERENCE?

**Comment by xxalpha** [comment]
>can be built in robotics and r&d

This isn't overpowered at all fam.

Stop misusing cyber implants for MORE GUNS. If you want robotic arms use the robotic augmentations.

**Comment by XDTM** [comment]
Stuff like chainsaw implants removes the use of the hand, which doesn't make too much sense if you're using a robotic arm. And using limbs instead of an organ would mean i'd have to remake a lot of procs.

**Comment by PKPenguin321** [comment]
Yeah I think this is better balanced if it entirely replaces your arm

**Comment by XDTM** [comment]
I feel like one minute cooldown, diamond plus high tech requirements, and actually needing a surgeon are enough limitations to be honest

**Comment by XDTM** [comment]
Also don't forget that any misclick wastes a charge

**Comment by Kromgar** [comment]
@XDTM hey whatever happened ot the rest of those lavaland organ prs by the way?

**Comment by xxalpha** [comment]
This is not what cyber implants were made for, this is exactly what robotic augmentations were made for.

**Comment by vcordie** [comment]
buffing rnd? reee

**Comment by XDTM** [comment]
@Kromgar i actually completely forgot, i guess i'll work on them now.
@xxalpha i'm fairly sure they were made to be metallic limbs without special functions.

**Comment by dannno** [comment]
embarrassing, unnecessary, not needed, pointless

**Comment by juanototo2** [comment]
>this piece of shit can murder deathsquads

Stop buffing RnD, you piece of shit.

**Comment by GunHog** [comment]
Make this item a cybernetic arm instead of an implant. A weapon this powerful needs such a drawback.

**Comment by ShadowDeath6** [comment]
you can only kill deathsquads(or anyone, really) if you hit them 3 times with it, since 40 + 40 is still less than 100.  You'd also have to hit them all three times and not miss any clicks or risk losing a charge since you'd end up pulling something else toward you rather than hitting the guy.

**Comment by XDTM** [comment]
I feel like you're kinda underestimating the cooldown. Also if you wanted to do damage you could just print an advanced egun, which is cheaper, ranged, has more charges and can stun, while not requiring implantation.

**Comment by XDTM** [comment]
I'll lower the damage to 30, which would make it an esword that can hit once a minute in average when used in combat, minus misclicks.
Before you cry "esword in rnd", count a minute and realize it's about 0.5 damage per second, again barring misclicks or other uses.

**Comment by xxalpha** [comment]
I made the cyber implants, they were not made to be limbs. They are IMPLANTS.

Fuck off with your number balancing memes to justify a bad feature.

**Comment by XDTM** [comment]
We already have toolkit and surgery IMPLANTS which work the same way, extending and retracting a NODROP item.

**Comment by xxalpha** [comment]
I don't care if you're lazy, make them robotic augs.

**Comment by XDTM** [comment]
That's less being lazy and more following the current standard for retractable surgical items. Special limbs are currently handled by NODROP items added via prosthetic replacement, but they completely remove the use of the hand.

**Comment by Armhulen** [comment]
bad feature

or as the danno says it
>embarrassing, unnecessary, not needed, pointless

**Comment by PKPenguin321** [comment]
It occurs to me that this is almost literally the same as changeling tentacles. Long cooldown with charges (chems for lings), pulls at range, large armor piercing damage once in melee, can steal items at range, etc

**Comment by XDTM** [comment]
They can both pull, although in a different way mechanically, cooldowns aren't exactly a rare thing, and the armor-piercing damage should never have been in the tentacles.

**Comment by PKPenguin321** [comment]
either way it's basically the same only it doesnt have a built in stun/disarm and instead deletes items outright

**Comment by PKPenguin321** [comment]
i think for now i'd be okay with adding this but not as an obtainable R&D thing unless it's remade into an actual arm replacement

**Comment by Armhulen** [comment]
or we could not? I don't want the wizard to have a changeling tentacle

**Comment by XDTM** [comment]
Is there a particular reason it can't be an implant? The difference is a slightly different surgery and not being able to retract it.

**Comment by PKPenguin321** [comment]
Yeah, it has the added tradeoff of being I not concealable and requiring the user to give up a hand to use it.

**Comment by optimumtact** [comment]
Well this went about as well as expected

**Comment by kevinz000** [comment]
poor R&D.

**Comment by AdamElTablawy** [comment]
this is kinda dumb tbh

i mean i understand that it has a long cooldown but remember that the majority of the time this is going to  go to the antagonist hunting skeleton changeling research director cruising around on his no-slowdown secway with his three advanced energy guns

do we really need this

**Comment by dannno** [comment]
nope.

**Comment by kevinz000** [comment]
yes

**Comment by optimumtact** [comment]
Basiclaly I think R&D already has heaps of op weapons and items so I don't know why people add more to them

It's a no from me

**Comment by XDTM** [comment]
My impression from testing is that it's actually not that powerful, especially considering the cost to produce. To compare, hulk gives the same machinery damage with no cooldown, the damage on mobs is high but can miss (and can at most deal 90 damage before having to wait 180 seconds). The pull is unique, but it has no stun and it's stopped by walls, so it won't keep anyone in melee for long, and item deletion is strong, but if you're powergaming you usually want to _steal_ your opponent's stuff.

I meant this as a gimmick more than as a weapon.

**Comment by Kromgar** [comment]
Make it a traitor item

**Comment by XDTM** [comment]
Traitor items are the new dumping ground for gimmick items and i don't like it as they're tied to one gamemode only

**Comment by Iamgoofball** [comment]
could be worse

it could be in cargo

**Comment by PKPenguin321** [comment]
Or worst of all, in the protolathe

**Comment by JJRcop** [comment]
What's holding this up?

**Comment by PKPenguin321** [comment]
Very few like it because it's basically changeling tentacle but for R&D

**Comment by GunHog** [comment]
>>Very few like it

>Twice the :+1: than :-1: on the PR

**Comment by KorPhaeron** [comment]
@optimumtact 

It might end up overpowered and stupid but as far as I know there is not a single melee weapon currently buildable in science, so it will at least be novel vs the piles of guns they make

File: code/modules/surgery/organs/augments_arms.dm [hunk 1]
```diff
@@ -236,3 +236,213 @@
...
+			if(QDELETED(A) || get_dist(A,user) > 12 || A.anchored) //breaks in case of teleportation
+				targets -= A
+				continue
+			if(!A.Move(get_step_towards(A, user)))
+				targets -= A
+				continue
+			if(get_dist(A,user) <= 1)
+				targets -= A
+				continue
+		sleep(1)
```
Comment (Iamgoofball) [inline code/modules/surgery/organs/augments_arms.dm]: no, check_tick

Comment (XDTM) [inline code/modules/surgery/organs/augments_arms.dm]: the delay is intentional to make the objects move smoothly, CHECK_TICK makes it instant

Comment (XDTM) [inline code/modules/surgery/organs/augments_arms.dm]: unless i don't know how to use check_tick properly

**Comment by PKPenguin321** [comment]
>Twice the 👍 than 👎 on the PR

Mobile github doesn't show these, leaving really only negative feedback in comments to view

**Comment by xxalpha** [comment]
Better to read comments than look at thoughtless upboats.

**Comment by XDTM** [comment]
The same person can make several comments though, which can be just as thoughtless

**Comment by ShadowDeath6** [comment]
>not using desktop mode on mobile

smh

**Comment by PKPenguin321** [comment]
>The same person can make several comments though, which can be just as thoughtless

The comments are from different people

**Comment by XDTM** [comment]
Each of which comments several times. A group of a few people can engage in a long discussion while several prefer to up/downvote instead of commenting their opinion, which can make the comment section skewed towards the most vocal group.

**Comment by PKPenguin321** [comment]
>i-it's the same three people talking



??????

**Comment by XDTM** [comment]
You're ten out of 37 non-mine comments, of course it's more visible.

Also this kinda went offtopic.

**Comment by optimumtact** [comment]
@tgstation/commit-access if nobody else decides what to do with this in two days I'm going to close it because personally I don't think it's a good change

**Comment by GunHog** [comment]
Please vote merge!

**Comment by Dorsisdwarf** [comment]
I don't like this. I just kept reading down the list of features and went "what" at each successive one.

**Comment by XDTM** [comment]
What's wrong with them?

**Comment by GunHog** [comment]
I think the Sci-haters could be reduced by about 16% if you made it an arm replacement.

**Comment by XDTM** [comment]
From what i've tested, that'd be worse than just having an arm. Once a minute, you get to delete something, pull something closer, or deal some damage, provided you don't misclick. I'd prefer to nerf other portions of it first, then scale it back after nobody ever builds it.

**Comment by KorPhaeron** [comment]
I keep meaning to merge this but I have been too lazy to review it sorry

**Comment by ShadowDeath6** [comment]
make this a holoparasite type and then add injectors to R&D :^)

**Comment by XDTM** [comment]
kor :(

**Pull request closed without merging**